pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -854,12 +855,9 @@ func (s *DiscoveryServer) startPush(version string, push *model.PushContext, ful
 		// indicates whether to do a full push for the proxy
 		proxyFull := full
 		if !full {
-			s.proxyUpdatesMutex.Lock()
-			if _, ok := s.proxyUpdates[client.modelNode.IPAddresses[0]]; ok {
+			if _, ok := proxyUpdates[client.modelNode.IPAddresses[0]]; ok {
 				proxyFull = true
-				delete(s.proxyUpdates, client.modelNode.IPAddresses[0])
```

Comment:
Before we just deleted this `client.modelNode.IPAddresses[0]` now we delete the whole cache, is this what we want?

Reply:
I want to make it looks like edsUpdates, clear the whole when a new push epoch start.  so `proxyUpdates` lifecycle is between two pushed.